### PR TITLE
Editor Improvemnets

### PR DESCRIPTION
**Pull request by RAVEENSR**
## Purpose
This PR contains editor improvements and bug fix.

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/commons/js/workspace/workspace.js [hunk 1]
```diff
@@ -313,11 +313,30 @@ define(['ace/ace', 'jquery', 'lodash', 'log','dialogs','./service-client','welco
...
+            this.updateDeleteMenuItem = function(){
+                var activeTab = app.tabController.getActiveTab(),
+                    deleteMenuItem = app.menuBar.getMenuItemByID('file.delete'),
+                    file = undefined;
+
+                if(activeTab.getTitle() != "welcome-page"){
+                    file = activeTab.getFile();
+                }
+
+                if(file !== undefined){
```
Comment (TanyaM) [inline components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/commons/js/workspace/workspace.js:331]: formatting issue?

File: components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/commons/js/workspace/workspace.js [hunk 2]
```diff
@@ -313,11 +313,30 @@ define(['ace/ace', 'jquery', 'lodash', 'log','dialogs','./service-client','welco
...
-                    deletesMenuItem.disable();
+                }
+            };
+
+            this.updateDeleteMenuItem = function(){
+                var activeTab = app.tabController.getActiveTab(),
+                    deleteMenuItem = app.menuBar.getMenuItemByID('file.delete'),
+                    file = undefined;
+
+                if(activeTab.getTitle() != "welcome-page"){
```
Comment (TanyaM) [inline components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/commons/js/workspace/workspace.js:327]: single quotes?

File: components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/commons/js/workspace/workspace.js [hunk 3]
```diff
@@ -313,11 +313,30 @@ define(['ace/ace', 'jquery', 'lodash', 'log','dialogs','./service-client','welco
...
+                    file = activeTab.getFile();
+                }
+
+                if(file !== undefined){
+                    file = activeTab.getFile();
+                    if(file.isPersisted()){
```
Comment (TanyaM) [inline components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/commons/js/workspace/workspace.js:333]: formatting issue?

File: components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/commons/js/workspace/workspace.js (hunk 1 above)
Comment (RAVEENSR) [reply components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/commons/js/workspace/workspace.js:331]: Fixed in 01a6ed2d65f1fdec1b6be164e9550899def3a82b

File: components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/commons/js/workspace/workspace.js (hunk 2 above)
Comment (RAVEENSR) [reply components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/commons/js/workspace/workspace.js:327]: Fixed in 01a6ed2d65f1fdec1b6be164e9550899def3a82b

File: components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/commons/js/workspace/workspace.js (hunk 3 above)
Comment (RAVEENSR) [reply components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/commons/js/workspace/workspace.js:333]: Fixed in 01a6ed2d65f1fdec1b6be164e9550899def3a82b

**Pull request merged**